src/main/java/com/github/princesslana/eriscasper/data/immutable/encoding/OptionalImmutableListEncoding.java

```diff
@@ -0,0 +1,48 @@
+package com.github.princesslana.eriscasper.data.immutable.encoding;
+
+import com.google.common.collect.ImmutableList;
+import java.util.Arrays;
+import java.util.Optional;
+import org.immutables.encode.Encoding;
+import org.immutables.encode.Encoding.StandardNaming;
+
+@Encoding
+public class OptionalImmutableListEncoding<T> {
```

Comment:
Why is this needed but NullableImmutableListEncoding isn't?

Reply:
Otherwise, it looks good.

Reply:
NullableImmutableListEncoding may be needed, but I haven't found a spot where it is yet. So, I haven't bothered with it for this PR.

Reply:
Gotcha